src/ape/managers/project/manager.py

```diff
@@ -95,6 +95,16 @@ def contracts_folder(self) -> Path:
 
         return folder
 
+    @property
+    def compiler_cache_folder(self) -> Path:
+        """
+        The path to the project's compiler source cache folder.
+        """
+        # NOTE: as long as config has come up properly, this should not be None
```

Comment:
idea: could return the default if it is None

Reply:
This error shouldn't happen under normal operation.  And I really don't want to do a second default that may conflict with the source of truth and also should never happen.

I'm actually a little more inclined to use an assert here rather an error with in a dead code path.  It was mostly to shut up the type checker.

Reply:
Oh, I see, ya because even if the user didn't configure (left as None), it would still be set after the first loading of the config.

An typing assert works, or maybe a builtin error like `raise TypeError("Expected non-None cache folder")`

all just a nit at this point

Reply:
Moved to an assert